openstack/tools/openstack_regression_tests_runner.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash -eu
+#
+# Run Openstack regression tests.
+#
+FUNC_TEST_PR=
+FUNC_TEST_TARGET=
+IMAGES_PATH=$HOME/tmp
+
+usage () {
+    cat << EOF
+USAGE: `basename $0` OPTIONS
+
+Run Openstack regression tests.
+
+OPTIONS:
+    --func-test-target TARGET_NAME
+        Provide the name of a specific test target to run.
+    --func-test-pr PR_ID
+        Provides similar functionality to Func-Test-Pr in commit message. Set
+        to zaza-openstack-tests Pull Request ID.
+    --help
+        This help message.
+EOF
+}
+
+while (($# > 0)); do
+    case "$1" in
+        --debug)
+            set -x
+            ;;
+        --func-test-target)
+            FUNC_TEST_TARGET=$2
+            shift
+            ;;
+        --func-test-pr)
+            FUNC_TEST_PR=$2
+            shift
+            ;;
+        --help|-h)
+            usage
+            exit 0
+            ;;
+        *)
+            echo "ERROR: invalid input '$1'"
+            usage
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+if [[ -z $FUNC_TEST_TARGET ]]; then
+   echo "ERROR: must provide a target name with --func-test-target"
+   exit 1
+fi
+
+# This is required for magnum tests and zaza will look in swift if it is not cached so we need to cache it first.
+mkdir -p $IMAGES_PATH
+if ! [[ -e $IMAGES_PATH/fedora-coreos-35.qcow2 ]]; then
+    wget https://builds.coreos.fedoraproject.org/prod/streams/stable/builds/35.20220424.3.0/x86_64/fedora-coreos-35.20220424.3.0-openstack.x86_64.qcow2.xz -O $IMAGES_PATH/fedora-coreos-35.qcow2.xz
+    (cd $IMAGES_PATH; xz -d fedora-coreos-35.qcow2.xz; )
+fi
+
+# Install dependencies
+which yq &>/dev/null || sudo snap install yq
+
+# Ensure zosci-config checked out and up-to-date
+(
+cd
+if [[ -d zosci-config ]]; then
+    cd zosci-config
+    git checkout master
+    git pull
+else
+    git clone https://github.com/openstack-charmers/zosci-config
+fi
+)
+
+TOOLS_PATH=$(realpath $(dirname $0))/func_test_tools
+CHARM_PATH=$(pwd)
+
+echo "Running regression tests"
+
+source ~/novarc
+export {,TEST_}CIDR_EXT=`openstack subnet show subnet_${OS_USERNAME}-psd-extra -c cidr -f value`
```

Comment:
Backticks are used in some places while `$(..)` are used in others. `$(..)` is generally recommended, so can stick to that for consistency as well.

See https://mywiki.wooledge.org/BashFAQ/082

Reply:
done. also did the same in charmed_openstack_functest_runner.sh